src/test/java/com/conveyal/gtfs/util/UtilTest.java

```diff
@@ -26,4 +30,34 @@ public void canHumanize() {
         assertThat(human(1234567), is("1.2M"));
         assertThat(human(1234567890), is("1.2G"));
     }
+
+    @Test
+    public void canEnsureValidNamespace() {
+        testNamespace("abc123", false);
+        testNamespace("abc_123", false);
+        testNamespace("abc_123.", false);
+        testNamespace("abc 123", true);
+        testNamespace("' OR 1=1;SELECT '1", true);
+    }
+
+    private void testNamespace(String namespace, boolean shouldFail) {
+        boolean errorThrown = false;
+        try {
+            ensureValidNamespace(namespace);
+        } catch (IllegalStateException e) {
+            errorThrown = true;
+            if (shouldFail) {
+                assertThat(e.getMessage(), equalTo(
+                    "Namespace must only have alphanumeric characters or the underscore symbol"
```

Comment:
This is testing for exact implementation details (the exact contents of an error message) instead of general behavior (it fails when input is not clean). We should just check for an exception, and possibly check the type of exception (in which case maybe we should define our own exception type, e.g. PossibleSQLInjectionException).

Another option is to perform all SQL injection testing with methods that return unambiguous booleans (acceptable / non-acceptable) instead of exceptions. Then we can just test that boolean value, which purely concentrates on the behavior and meaning instead of implementation.